src/Campaigns/Repositories/CampaignRepository.php

```diff
@@ -125,6 +145,30 @@ public function update(Campaign $campaign): void
                     'start_date' => $startDateFormatted,
                     'end_date' => $endDateFormatted,
                 ]);
+
+            if ( ! is_null($donationForm)) {
```

Comment:
The query you mentioned only updates the values, but we can't insert DB entries with that....

I was able the replace the queries for insert/delete with a single one that can insert or update value in case of duplication. 

But I still need an additional query for cases where the $isDefault is true to make sure that we'll keep only one form as default on DB.

Resolved in af83cc387557764bd1b8c1e60417b72f58178ced